analyzer/runtime/evm/client.go

```diff
@@ -313,21 +313,30 @@ func EVMMaybeUnmarshalEncryptedData(data []byte, result *[]byte) (*EVMEncryptedD
 		if err := cbor.Unmarshal(*result, &callResult); err != nil {
 			return nil, fmt.Errorf("unmarshal outer call result: %w", err)
 		}
-		if callResult.IsUnknown() {
-			switch call.Format {
-			case sdkTypes.CallFormatEncryptedX25519DeoxysII:
-				var resultEnvelope sdkTypes.ResultEnvelopeX25519DeoxysII
+		switch call.Format {
+		case sdkTypes.CallFormatEncryptedX25519DeoxysII:
+			var resultEnvelope sdkTypes.ResultEnvelopeX25519DeoxysII
+			switch {
+			// The result is encrypted.
+			case callResult.IsUnknown():
 				if err := cbor.Unmarshal(callResult.Unknown, &resultEnvelope); err != nil {
-					return nil, fmt.Errorf("outer call result unmarshal unknown: %w", err)
+					return nil, fmt.Errorf("outer call result unmarshal ok: %w", err)
+				}
+			// The result can be unencrypted if the input was unencrypted.
+			case callResult.IsSuccess() && !callResult.IsUnknown():
+				if err := cbor.Unmarshal(callResult.Ok, &resultEnvelope); err != nil {
+					return nil, fmt.Errorf("outer call result unmarshal ok: %w", err)
 				}
-				encryptedData.ResultNonce = resultEnvelope.Nonce[:]
-				encryptedData.ResultData = resultEnvelope.Data
 			default:
-				// We have already checked when decoding the call envelope,
-				// but I'm keeping this default case here so we don't forget
-				// if this code gets restructured.
-				return nil, fmt.Errorf("outer call format %s (%d) not supported", call.Format, call.Format)
+				return nil, fmt.Errorf("inner callResult failed")
 			}
+			encryptedData.ResultNonce = resultEnvelope.Nonce[:]
+			encryptedData.ResultData = resultEnvelope.Data
+		default:
+			// We have already checked when decoding the call envelope,
+			// but I'm keeping this default case here so we don't forget
+			// if this code gets restructured.
+			return nil, fmt.Errorf("outer call format %s (%d) not supported", call.Format, call.Format)
```

Comment:
Heads up: I've noticed this error in logs quite a bit:
```
{
analyzer: "sapphire"
caller: "extract.go:499"
err: "outer call format plain (0) not supported"
level: "error"
module: "analysis_service"
msg: "error unmarshalling encrypted data and result, omitting encrypted fields"
}
```

The `caller` above is a line in `extract.go` that calls the method from this PR (`EVMMaybeUnmarshalEncryptedData`). So basically, we've got txs with `call.Format == sdkTypes.CallFormatPlain`, and we're not handling them. (This function generates the same error message twice, once here and once in line 309.)
I haven't created a ticket for this IIRC, but I'm wondering if it's on your radar already since you've been looking at encryption a lot lately? It's probably worth handling in the next days as you're working in this area.

Reply:
Thanks for raising this! I think in this case, the analyzer is working as expected since we aren't able to extract `evm_encrypted_*` fields from a plain tx. I'll add an explicit switch case to reduce the log spam here though.

The one equivalent field we _could_ extract is the `data_data`, eg the evm call data. However, it likely requires its own dedicated column, and also we currently store that as part of the `body -> data`.

Reply:
wait what do you mean by "twice" and "log spam"? it shows up in the code lexically twice, but they return when either one happens

Reply:
Hm so when the outer call format is `plain (0)`, do we still want to log an error? My original idea was to simply return no error in this case, since the docstring says that `If the data is not encrypted,it returns nil with no error.`. I might be missing something though. If the callformat is 0, does it guarantee that the transaction is not encrypted?

Reply:
"if the data is not encrypted" here refers to a certain thing about how Sapphire style encryption is. the data field of an otherwise unencrypted Ethereum format transaction is a valid CBOR-encoded Oasis `Call` structure (the hope is that no contract will incidentally expect anything compatible as the actual data, initcode for "create" as well).

so the phrase in the description (as well as the "Maybe" in the name) there refers to handling this ambiguity. it's only to say you can pass in whatever the data/initcode you get and it'll do the right thing.

as for how plain (0) should fit in, imo return nil with no error. the goal of this function is to populate the encrypted envelope fields for insertion into the database. and the change in this PR to look for the encrypted envelope in callResult.Ok is correct too.

Reply:
addendum: now that I think about it, maybe it's more suitable to handle invalid `result` CBOR with similar logic inside this function instead of special-casing the code around it to pass `nil` for "create."

Reply:
yup!